x-pack/solutions/security/plugins/security_solution/public/common/lib/integrations/types/index.ts

```diff
@@ -5,11 +5,12 @@
  * 2.0.
  */
 import type { AvailablePackagesHookType, IntegrationCardItem } from '@kbn/fleet-plugin/public';
+import type { GetInstalledPackagesResponse } from '@kbn/fleet-plugin/common/types';
 import type { UseSelectedTabReturn } from '../hooks/use_selected_tab';
 
 export interface IntegrationCardMetadata {
-  installedIntegrationsCount: number;
-  isAgentRequired: boolean;
+  isAgentRequired?: boolean;
```

Comment:
The `isAgentRequired` prop seems to always be set in the check_complete. Is there a reason to make it optional?

Reply:
There are some featured integrations in search ai lakes are [agentless](https://www.elastic.co/docs/solutions/security/get-started/agentless-integrations#beta-agentless-integrations)
so here it does not check if isAgentRequired, but still shares the same `IntegrationCardMetadata`.  x-pack/solutions/security/plugins/security_solution/public/onboarding/components/onboarding_body/cards/integrations_external_detections/integrations_check_complete.ts .
I add a new type `ExternalIntegrationCardMetadata` to distinguish between the two, wdyt?

Reply:
I see. yup! that makes a lot of sense 👍